### ISSUE TITLE
Regarding similarity between training and test datasets 

### ISSUE THREAD

**Issue by sudharavali** [junyanz/pytorch-CycleGAN-and-pix2pix#634]
I just have a question regarding the training and test data of CycleGan. 

Do you think the test data must belong to the same domain as that of the training data(for synthetic dataset), when I mean domain, I am talking about the backgrounds both of them share. When I give the test data as the drone with different background, its not producing the same quality of results as it does when given a test image with a drone that shares similar background as seen in the training dataset. 
Is this result correct, should we make sure that the backgrounds should remain as similar to each other as possible ? 
Please let me know. 

The result obtained when the background is same : 
![Drone_1002 (8)_fake_B](https://user-images.githubusercontent.com/33206551/57030993-691a7c80-6c14-11e9-8474-3f6858f5e0c5.png)

the result obtained when the background is different : 

![file-1_fake_B](https://user-images.githubusercontent.com/33206551/57030927-40928280-6c14-11e9-84d0-34723d46e67e.png)

Input synthetic image (trainA dataset) :

![Drone_1003 (7)_real_A](https://user-images.githubusercontent.com/33206551/57031040-8b13ff00-6c14-11e9-8f55-25f90b43c5cf.png)

**Comment by junyanz** [comment]
You are correct. The test data should be similar to training data. I recommend that you collect additional training data or apply additional data augmentation. (e.g., different kinds of cropping/scaling)

**Comment by sudharavali** [comment]
Thank you so much for the prompt response Professor. I just have another follow up question, when do you think I can stop the training process ? Is human involvement/ observation required  or are there any other methods to do it like keeping track of any particular loss function at their minimum , etc ? 
Please let me know. Thank you .

**Comment by junyanz** [comment]
If you have a downstream task, you can evaluate the performance of your model regarding the task. Otherwise, it requires either (1) manual inspection to choose the best model, or (2) standard GAN metrics (e..g, [FID](https://github.com/mseitzer/pytorch-fid))

**Comment by sudharavali** [comment]
Thanks alot Professor. This clarifies alot of questions I had.

**Comment by prannerta100** [comment]
Hi, I just started playing with the horse2zebra dataset. I am a little new to GANs. In usual ML models, I am used to training data having a one-one correspondence. That is, if horse pics imageA001 to imageA009.jpg are in folder trainA, then imageB001 to imageB009.jpg should be the corresponding zebra images with the same background and so on, just the horse body replaced with a ditto zebra body. But then I don't see this kind of a one-one labeling in the train data folders. Is such a one-one labelling unimportant for GANs?

**Comment by junyanz** [comment]
It depends on the type of GAN models. For CycleGAN, we don't need one-to-one correspondence. For pix2pix, we need.

**Comment by prannerta100** [comment]
Thanks!

In your tutorial ipynb, I ran the following command to train my model (I know n_epochs is too small, but this is more of a sanity check whether things work, as each epoch takes 400 s on my Google Colab single GPU).
 
`!python train.py --dataroot ./datasets/horse2zebra --name horse2zebra --model cycle_gan --gpu_ids 0 --n_epochs 1 --n_epochs_decay 1  --display_id 0`

This is your horse2zebra dataset, nothing new from my side.

But then I don't generate `./checkpoints/horse2zebra/latest_net_G_A.pth` as expected. In fact there is no .pth file generated. Whereas the pretrained folder `./checkpoints/horse2zebra_pretrained` has this pth file.

**Comment by junyanz** [comment]
By default, your models will be saved every `--save_latest_freq` iterations (default 5000) or every `--save_epoch_freq` (default 5) epoches. In your case, as you only trained your model for 2 epoches, none of them would be saved.
